code/obj/sawflymisc.dm

```diff
@@ -0,0 +1,581 @@
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to sprial out of contro!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+	proc/CritterDeath() // rip lil guy
+
+		if (!src.alive) return // can't double-die
+		alive = FALSE
+		src.tokenized_message(death_text)
+		src.is_npc = FALSE // stop any and all possible non-critter AI thought
+		SEND_GLOBAL_SIGNAL(COMSIG_GLOBAL_DRONE_DEATH, src)
+
+		if (prob(20))
+			new /obj/item/device/prox_sensor(src.loc) // maybe change this later
+			return
+
+
+		/*SHOULD_CALL_PARENT(TRUE) //commenting this out because I'm afraid everything will break if I delete it
+		#ifdef COMSIG_OBJ_CRITTER_DEATH
+		SEND_SIGNAL(src, COMSIG_OBJ_CRITTER_DEATH)
+		#endif*/
+
+
+		//death behavior custom to sawflies below
+		animate(src) //stop no more float animation
+		icon_state = "sawflydead[pick("1", "2", "3", "4", "5", "6", "7", "8")]"
+		src.pixel_x += rand(-5, 5)
+		src.pixel_y += rand(-1, 5)
+
+		src.alive = 0
+		src.anchored = 0
+		src.set_density(0)
+		walk_to(src,0) //halt walking
+
+		// special checks that determine how much postmorten chaos our little sawflies cause
+
+		if(prob(60)) // 60 percent chance to zap the area
+			elecflash(src, 1, 2)
+
+		if(prob(20)) // congrats, little guy! You're special! You're going to blow up!
+			if(prob(70)) //decide whether or not people get a warning
+				src.visible_message("<span class='combat'>[src] makes a [pick("gentle", "odd", "slight", "weird", "barely audible", "concerning", "quiet")] [pick("hiss", "drone", "whir", "thump", "grinding sound", "creak", "buzz", "khunk")].......")
+			SPAWN(deathtimer SECONDS) // pause, for dramatic effect
+				src.blowup()
+
+
+	proc/blowup() // used in emagged controllers and has a chance to activate when they die
+
+		if(prob(66))
+			src.visible_message("<span class='combat'>[src]'s [pick("motor", "core", "fuel tank", "battery", "thruster")] [pick("combusts", "catches on fire", "ignites", "lights up", "bursts into flames")]!")
+			fireflash(src,1,TRUE)
+		else
+			src.visible_message("<span class='combat'>[src]'s [pick("motor", "core", "head", "engine", "thruster")] [pick("overloads", "blows up", "catastrophically fails", "explodes")]!")
+			fireflash(src,0,TRUE)
+			explosion(src, get_turf(src), 0, 0.75, 1.5, 3)
+			qdel(src)
+
+		if(alive) // prevents weirdness from emagged controllers causing frankenstein sawflies
+			qdel(src)
+
+
+
+	death() //FUCK YOU
+		CritterDeath() // FUCK YOU. AND MOST CERTAINLY,
+		alive = FALSE // FUUUUUUCK. YOUUUUUUU.
+
+
+	// Ported critter AI stuff from hereon out
+
+
+
+	proc/ChaseAttack(atom/M) // overriding these attack procs so drone is nicer to traitors >:(
+		if (istraitor(M) || isnukeop(M) || isspythief(M) || (M in src.friends))
+			return
+		if(target && !attacking)
+			attacking = 1
+			src.visible_message("<span class='alert'><b>[src]</b> flies at [M]!</span>")
+			if (istraitor(M) || isnukeop(M) || isspythief(M) || (M in src.friends))
```

Comment:
Why is this check here again?

Reply:
It ensues that the message is displayed, but the sawfly doesn't actually go after the traitor, so it's less obvious who their master is.